core/src/wsv.rs

```diff
@@ -807,6 +828,23 @@ impl<W: WorldTrait> WorldStateView<W> {
     {
         self.modify_world(|world| f(&world.triggers).map(WorldEvent::Trigger))
     }
+
+    /// Execute trigger with `trigger_id` as id and `authority` as owner
+    ///
+    /// Produces [`ExecuteTriggerEvent`].
+    /// Actual trigger execution will be in the next [`Self::apply()`] call
```

Comment:
It seems to me that within the same `apply()`, `ExecuteTriggerEvent` is picked up immediately after `execute_transactions()` and can pull the trigger

Reply:
Indeed, I've missed it. So there are two ways:

1. When `ExecuteTrigger` ISI is inside a *transaction*, *trigger* will be executed on the **same** block
2. When `ExecuteTrigger` ISI is inside a *trigger*, *trigger* will be executed on the **next** block

Is it okay for our purpose?

Reply:
Yes. I’ll just need to add the information to the tutorial.

Reply:
I'll mention this information in doc-comments and close this conversation